### PR TITLE
feat: add go pkg support

### PR DESCRIPTION
**Pull request by yingziwu**
add go pkg support

### CODE REVIEW

**Comment by lilydjwg** [comment]
Doesn't it have an API?

**Comment by yingziwu** [comment]
> Doesn't it have an API?

`pkg.go.dev` doesn't have the API.

See https://github.com/golang/go/issues/36785

File: nvchecker_source/go.py [hunk 1]
```diff
@@ -33,4 +33,4 @@ async def get_version_impl(info) -> VersionResult:
         version = elements[0]
         return version
     except IndexError:
-        raise GetVersionError(pkg_name, "parse error.")
```
Comment (lilydjwg) [inline nvchecker_source/go.py]: No, it's not the intended way. It should be

```
GetVersionError("parse error", name=pkg_name)
```

Comment (yingziwu) [reply nvchecker_source/go.py]: Thanks
It has been fixed.

**Pull request merged**